src/darwin/Framework/CHIP/MTRDeviceController.mm

```diff
@@ -531,6 +538,31 @@ - (BOOL)getBaseDevice:(uint64_t)deviceID
                  }];
 }
 
+- (MTRDevice *)deviceForDeviceID:(uint64_t)deviceID
+{
+    os_unfair_lock_lock(&_deviceMapLock);
+    MTRDevice * deviceToReturn = self.deviceIDToDeviceMap[@(deviceID)];
+    if (deviceToReturn) {
+        deviceToReturn = [[MTRDevice alloc] initWithDeviceID:deviceID deviceController:self queue:self.chipWorkQueue];
```

Comment:
The queue passed to MTRDevice is for its internal use, not for calling into client code. This is to align framework code on the same queue to avoid thread explosion.

All callbacks to client code in MTRDevice use queues that the clients provide.

Reply:
Ok, I was wondering.  ;)

The other places this pops up are:
* `connectAndPerformAsync` running the block on the MTRDevice's queue.  But I am wondering whether we can eliminate most uses of that and then re-evaluate.
* `readAttributeWithEndpointId` running the `MTRAsyncCallbackQueueWorkItem` on `_queue`.  This ends up invoking various code on the Matter queue, but I _think_ it turns out none of that code does a `dispatch_sync` to the Matter queue, so we're ok in practice.

That said, I think we need to do something to better keep track of which things are safe to call on the Matter queue vs safe to call on a non-Matter queue vs safe to call on any queue....